src/pkg/services/m365/m365.go

```diff
@@ -61,12 +65,87 @@ func UsersCompat(ctx context.Context, acct account.Account) ([]*User, error) {
 func UsersCompatNoInfo(ctx context.Context, acct account.Account) ([]*UserNoInfo, error) {
 	errs := fault.New(true)
 
-	users, err := usersNoInfo(ctx, acct, errs)
+	usersResult, err := usersNoInfo(ctx, acct, errs)
 	if err != nil {
 		return nil, err
 	}
 
-	return users, errs.Failure()
+	return usersResult, errs.Failure()
+}
+
+// UserHasMailbox returns true if the user has any mailbox enabled
+// false otherwise, and a nil pointer and an error in case of error
+func UserHasMailbox(ctx context.Context, acct account.Account, userID string) (*bool, error) {
+	uapi, err := makeUserAPI(acct)
+	result := false
+
+	if err != nil {
+		return nil, clues.Wrap(err, "getting mailbox").WithClues(ctx)
+	}
```

Comment:
styling: please prefer colocating errors with the function call.  Also, default valued variable declaration should use the format `var foo type`
```suggestion
	var result bool

        uapi, err := makeUserAPI(acct)
	if err != nil {
		return nil, clues.Wrap(err, "getting mailbox").WithClues(ctx)
	}
```